### DISCUSSION TITLE
What is the best way to transcribe a conversation?

### DISCUSSION THREAD

**Discussion by fanyangxyz** [openai/whisper#1191] [Q&A] [answered]
I have a recording of a therapy session, about 30-min conversation between a patient and her therapist. I used whisper to transcribe but the result is a long blob of text, not in the dialog format. I also tried using the prompt of "This is a conversation between ...". But it did not change the result much. Is whisper capable of transcribing conversation into dialog format? By dialog format I mean something like 
"Person A: ..."
"Person B: ..."

**Comment by Tex2002ans** [comment, marked as answer]
> I used whisper to transcribe but the result is a long blob of text, not in the dialog format. [...] Is whisper capable of transcribing conversation into dialog format? 
>
> "Person A: ..."
> "Person B: ..."

No. Whisper can't currently do this.

The technical term for this is called:

- Speaker Diarization.

You can find discussion on this in other topics:

- https://github.com/openai/whisper/discussions/104
- https://github.com/openai/whisper/discussions/264
- https://github.com/openai/whisper/discussions?discussions_q=is%3Aopen+diarization

Currently, the closest answer was given by jongwook:

- https://github.com/openai/whisper/discussions/117#discussioncomment-3727051

where you can "hack" `initial_prompt` with hyphens, in order to nudge Whisper to potentially output dashes between speakers.

> [...] You can also use this for "prompt engineering", to inform the model to become more likely to output certain jargon (`" So we were just talking about DALL·E"`) or **do a crude form of speaker turn tracking (e.g. `" - Hey how are you doing? - I'm doing good. How are you?"`, note that the token for `" -"` is suppressed by default and will need to be enabled manually.)**

Besides that, you may need to use/program external tools—like Pyannote—to try to take:

- Original Audio + Whisper's transcripts

which will generate timings, which would help tag certain lines as Speaker A + Speaker B.

(See those previously linked topics for much more details.)

**Reply by fanyangxyz** [reply]
Thanks so much for the information. I'll look into them.

**Reply by Majdoddin** [reply]
@fanyangxyz @Tex2002ans 
www.lexicaps.com adds diarization to Whispers transcription.  No 3rd party packages.
Announcement: #1537 
Repo: https://github.com/Majdoddin/lexicaps

**Comment by Frosti7** [comment]
Yes, it does support speaker

   - Speaker Diarization.

https://github.com/lablab-ai/Whisper-transcription_and_diarization-speaker-identification-

**Comment by Joshfindit** [comment]
Interested to see if this has changed with the release of Whisper 3

**Reply by envious** [reply]
Same
